File: frontend/communityai/src/features/inbox/types/inbox.ts
export type InboxMessageType = 'COMMENT' | 'MESSAGE' | 'MENTION'

export type InboxSentiment = 'POSITIVE' | 'NEUTRAL' | 'NEGATIVE' | 'UNKNOWN'

export interface SocialAccountSummary {
  id: number
  platform: string
  provider: string
  account_name: string
  account_username?: string | null
  profile_image_url?: string | null
}

export interface InboxMessage {
  id: number
  user_id: number
  social_account_id: number
  external_id: string
  type: InboxMessageType
  sender_name: string
  sender_external_id?: string | null
  content: string
  sentiment: InboxSentiment
  sentiment_score?: number | null
  is_read: boolean
  is_resolved: boolean
  replied_at?: string | null
  created_at: string
  updated_at: string
  social_account?: SocialAccountSummary | null
}

export interface InboxListResponse {
  items: InboxMessage[]
  total: number
  unread_count: number
}

export interface InboxUnreadCountResponse {
  unread_count: number
}

export interface InboxFiltersState {
  type?: InboxMessageType | 'ALL'
  platform?: string
  sentiment?: InboxSentiment | 'ALL'
  status?: 'ALL' | 'UNREAD' | 'READ' | 'RESOLVED'
  search?: string
}

export interface InboxReplyRequest {
  content: string
}

export interface InboxSuggestReplyRequest {
  tone?: string
  instructions?: string
}

File: backend/app/schemas/inbox.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.inbox_message import InboxMessageType, InboxSentiment
from app.schemas.ai import AITone


class SocialAccountSummary(BaseModel):
    id: int
    platform: str
    provider: str
    account_name: str
    account_username: str | None = None
    profile_image_url: str | None = None

    model_config = ConfigDict(from_attributes=True)


class InboxMessageResponse(BaseModel):
    id: int
    user_id: int
    social_account_id: int
    external_id: str
    type: InboxMessageType
    sender_name: str
    sender_external_id: str | None = None
    content: str
    sentiment: InboxSentiment
    sentiment_score: float | None = None
    is_read: bool
    is_resolved: bool
    replied_at: datetime | None = None
    created_at: datetime
    updated_at: datetime
    social_account: SocialAccountSummary | None = None

    model_config = ConfigDict(from_attributes=True)


class InboxListResponse(BaseModel):
    items: list[InboxMessageResponse]
    total: int
    unread_count: int


class InboxUnreadCountResponse(BaseModel):
    unread_count: int


class InboxMarkReadRequest(BaseModel):
    is_read: bool = True


class InboxMarkResolvedRequest(BaseModel):
    is_resolved: bool = True


class InboxReplyRequest(BaseModel):
    content: str = Field(..., min_length=1, max_length=5000, description="Reply text content")


class InboxSuggestReplyRequest(BaseModel):
    tone: AITone | None = None
    instructions: str | None = Field(default=None, max_length=1000)


class InboxMessageCreate(BaseModel):
    user_id: int
    social_account_id: int
    external_id: str
    type: InboxMessageType
    sender_name: str
    sender_external_id: str | None = None
    content: str
    sentiment: InboxSentiment = InboxSentiment.UNKNOWN
    sentiment_score: float | None = None
    is_read: bool = False
    is_resolved: bool = False
